ui/app/services/version.js

```diff
@@ -48,7 +48,14 @@ export default class VersionService extends Service {
   }
 
   get hasSecretsSync() {
-    return this.features.includes('Secrets Sync');
+    const isEnterprise = this.isEnterprise;
+    const isHvdManaged = this.flags.isHvdManaged;
+    const onLicense = this.features.includes('Secrets Sync');
+
+    if (!isEnterprise) return false;
+    if (isHvdManaged) return true;
+    if (isEnterprise && !onLicense) return false;
+    return true;
```

Comment:
This logic is mostly copy/pasted from the flags service. The biggest change here is there's no check for activation and subsequently nav permissions. Neither of these felt relevant to answering "does this feature exist for this cluster" which is (now) what this check is responsible for.  

Now anytime `hasSecretsSync` is referenced if a permissions or a activation check is also necessary that happens as well.